functions/Get-DbaUserPermission.ps1

```diff
@@ -221,7 +221,12 @@ function Get-DbaUserPermission {
             foreach ($db in $dbs) {
                 Write-Message -Level Verbose -Message "Processing $db on $instance"
 
-                $db.ExecuteNonQuery($endSQL)
+                try {
+                    $db.ExecuteNonQuery($endSQL)
+                } catch {
+                    # here to avoid empty catch
+                    $null = 1
+                } # this will fail if the database is not accessible (i.e. restoring etc.)
```

Comment:
How can this be if you filter those dbs out in line 203?

Reply:
🤦‍♂️ I did not mean to leave that still in there: just pulled it back out.

Reply:
🤦‍♂️ I meant to pull all of that back out: just removed the last of my original "fix"

Reply:
Excuse me for being so clear: You removed a good solution and introduced a nasty litte try-catch-fix just to suppress an error and not really solve the problem. There are issues with this command and we sould solve (!) them in a good way. Perhaps I represent a minority opinion, let's see what the other contributors say...

Reply:
I can certainly go the other direction: I just took my cues from the existing code...

Reply:
Okay, I took a stab at cleaning it up. As a very new contributor I didn't want to assume I should do more extensive re-writes. This current version moves the server level permissions to a single check (without the need for a "did we do it already flag" at the beginning of the loop on `$sqlInstances` and then moves the check to see if a db is accessible into the loop on `$dbs` and displaying a warning message if the database is not queryable...

Reply:
@andreasjordan Thanks for the positive feedback that my initial changes were on the right track. I'm just super good at second guessing myself...